src/runtime_src/core/common/api/xrt_bo.cpp

```diff
@@ -122,6 +127,49 @@ send_exception_message(const std::string& msg)
 
 namespace xrt {
 
+// class bo::async_handle_impl - Base class for asynchronous buffer DMA handle
+//
+// Derived classes:
+// [aie::b ::async_handle_impl]: For AIE BOs
+//
+// Impl Class associated with async bo which allows to wait for completion
+class bo::async_handle_impl
+{
+public:
+  xrt::bo m_bo;
+
+public:
+  // async_bo_impl() - Construct async_bo_obj
+  async_handle_impl(xrt::bo bo)
+    : m_bo(std::move(bo))
+  {
+  }
+
+  // wait() - Wait for async to complete
+  virtual void 
+  wait();
+};
+
+class aie::bo::async_handle_impl : public xrt::bo::async_handle_impl
+{
+public:
+  size_t m_bd_num; // For future use
+  std::string m_gmio_name;
+
+public:
+  // async_bo_impl() - Construct async_bo_obj
+  async_handle_impl(xrt::bo bo, size_t bd_num, std::string gmio_name)
+    : xrt::bo::async_handle_impl(std::move(bo)),
+      m_bd_num(bd_num),
+      m_gmio_name(std::move(gmio_name))
+  {
+  }
```

Comment:
```suggestion
  {}
```
Clearly shows you didn't forget anything.

Reply:
Agree